eng/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -154,45 +185,83 @@ function ProcessExcludes([Array]$matrix, [Array]$excludes)
     return $matrix | Where-Object { !$_.parameters.Contains($deleteKey) }
 }
 
-function ProcessIncludes([Array]$matrix, [Array]$includes, [Hashtable]$displayNamesLookup)
+function ProcessIncludes([MatrixConfig]$config, [Array]$matrix)
 {
-    foreach ($inclusion in $includes) {
-        $converted = ConvertToMatrixArrayFormat $inclusion
-        $full = GenerateFullMatrix $converted $displayNamesLookup
-        $matrix += $full
+    $inclusionMatrix = @()
+    foreach ($inclusion in $config.include) {
+        $full = GenerateFullMatrix $inclusion $config.displayNamesLookup
+        $inclusionMatrix += $full
     }
 
-    return $matrix
+    return $matrix + $inclusionMatrix
 }
 
-function MatrixElementMatch([System.Collections.Specialized.OrderedDictionary]$source, [System.Collections.Specialized.OrderedDictionary]$target)
+function ProcessImport([System.Collections.Specialized.OrderedDictionary]$matrix, [String]$selection)
 {
-    if ($target.Count -eq 0) {
-        return $false
+    if (-not $matrix -or -not $matrix.Contains($IMPORT_KEYWORD)) {
+        return $matrix
     }
 
-    foreach ($key in $target.Keys) {
-        if (-not $source.Contains($key) -or $source[$key] -ne $target[$key]) {
-            return $false
+    $importPath = $matrix[$IMPORT_KEYWORD]
+    $matrix.Remove($IMPORT_KEYWORD)
+
+    $matrixConfig = GetMatrixConfigFromJson (Get-Content $importPath)
+    $importedMatrix = GenerateMatrix $matrixConfig $selection
+
+    return $matrix, $importedMatrix
+}
+
+function CombineMatrices([Array]$matrix1, [Array]$matrix2)
+{
+    $combined = @()
+    if (-not $matrix1) {
+        return $matrix2
+    }
+    if (-not $matrix2) {
+        return $matrix1
+    }
+
+    foreach ($entry1 in $matrix1) {
+        foreach ($entry2 in $matrix2) {
+            $newEntry = @{
+                name = $entry1.name
+                parameters = CloneOrderedDictionary $entry1.parameters
+            }
+            foreach($param in $entry2.parameters.GetEnumerator()) {
+                if (-not $newEntry.Contains($param.Name)) {
+                    $newEntry.parameters[$param.Name] = $param.Value
+                } else {
+                    Write-Warning "Skipping duplicate parameter `"$($param.Name)`" when combining matrix."
```

Comment:
I wonder if we should instead of skipping add the new values if any?

Reply:
This has the effect of favoring top level keys over imported keys, which I think is the right behavior. We can't have both.

Reply:
Or are you saying include all combinations with both?

Reply:
I was suggesting including all values from both sets.

Reply:
I think it makes more sense to have them combine as you suggest, though I'm inclined to do it as a follow-on change since the code might get complicated.